stockroom/storages/model.py

```diff
@@ -0,0 +1,120 @@
+import warnings
+
+import numpy as np
+
+from .. import parser
+from ..repository import StockRepository
+from ..utils import LazyLoader
+
+torch = LazyLoader('torch', globals(), 'torch')
+tf = LazyLoader('tf', globals(), 'tensorflow')
+
+
+class ModelStore:
+    """
+    ModelStore class utilizes hangar arraysets to store the pieces of a model and
+    use hangar metadata to store the information required to collate it back to
+    a model. It right now supports ``keras.Model`` and ``torch.nn.Module`` models.
+    ModelStore instance, on :meth:`storages.ModelStore.save` creates few arraysets
+    (one arrayset for each data type) to store the weights and create one arrayset
+    specifically to store the shape of each layer. This shape arrayset is needed
+    because the weights of each layer would be flattened before saving. This is
+    essential since handling variable shapes and variable ranks are more complex
+    than flattening and reshaping-back the weights.
+    """
+
+    def __init__(self, write):
+        self.co = None
+        self._write = write
+        self._repo = StockRepository()
+
+    def save(self, name, model):
+        if not self._write:
+            raise PermissionError("ModelStore instance is not write-enabled")
+        if hasattr(model, 'state_dict'):
+            state = model.state_dict()
+            weights = [x.numpy() for x in state.values()]
+            layers = state.keys()
+            library = 'torch'
+            library_version = torch.__version__
+        elif hasattr(model, 'get_weights'):
+            library = 'tf'
+            layers = None
+            weights = model.get_weights()
+            library_version = tf.__version__
+        else:
+            raise TypeError("Unknown model type. StockRoom can work with only "
+                            "``Keras.Model`` or ``torch.nn.Module`` modules")
+        longest = max([len(x.reshape(-1)) for x in weights])
+        dtypes = [w.dtype.name for w in weights]
+
+        co = self._repo.checkout(write=True)
```

Comment:
use a context manager. if any error occurs below then the checkout will not be closed.